src/lib/Lib/CMakeExecution.cpp

```diff
@@ -104,8 +131,54 @@ getCmakeDefaultGenerator(llvm::StringRef cmakePath)
     return Unexpected(Error("Default CMake generator not found"));
 }
 
+Expected<std::string>
+parseCmakeSystemInformationOutput(std::string const& cmakeSystemInformation)
+{
+    std::istringstream stream(cmakeSystemInformation);
+    std::string line;
+    std::string defaultGenerator;
+
+    while (std::getline(stream, line))
+    {
+        if (line.starts_with("CMAKE_GENERATOR \""))
+        {
+            size_t const start = 17;
+            size_t const end = line.find("\"", start);
+            if (end == std::string::npos)
+            {
+                continue;
+            }
+            return line.substr(start, end - start);
+        }
+    }
+    return Unexpected(Error("Default CMake generator not found"));
+}
+
+Expected<std::string>
+getCmakeDefaultGenerator(llvm::StringRef cmakePath)
+{
+    Expected<std::string> const cmakeHelpExp = executeCmakeHelp(cmakePath);
+    if (!cmakeHelpExp)
```

Comment:
In this case, it's easier to define the case where `cmakeHelpExp` is true first.

```cpp
if (cmakeHelpExp)
{
     return parseCmakeHelpOutput(*cmakeHelpExp);
}
// fall back to system information
}